src/js/providers/tracks-mixin.js

```diff
@@ -551,10 +551,7 @@ define(['utils/underscore',
             // already have one with the same label
             track = _.findWhere(tracks, { label: label });
 
-            if (track) {
-                track.kind = itemTrack.kind;
-                track.language = itemTrack.language || '';
-            } else {
```

Comment:
@johnBartos @robwalch We set `kind` and `language` to keep things consistent, but neither property is used visually, so it should be ok that they're not mutated when reusing a track. `label` is what ends up in the menu. It's likely that the `language` will be the same if the `label` hasn't changed and `kind` should only ever be `subtitles` or `captions` (never `metadata`).

+1 to needing `removeTextTrack`!